File: repl/adrianus/java/ad/src/main/java/com/ad/hash/HashPair.java
package com.ad.hash;

import com.ad.objects.AdObject;

public class HashPair<T> {
    private T key;
    private T value;

    public HashPair(T k, T v) {
        key = k;
        value = v;
    }

    public T getKey() {
        return key;
    }

    public void setKey(T key) {
        this.key = key;
    }

    public T getValue() {
        return value;
    }

    public void setValue(T value) {
        this.value = value;
    }
}

File: repl/adrianus/cpp-v3/hashpair.cpp
#include "hashpair.h"

HashPair::HashPair() {
	//key = NULL;
	//value = NULL;
}

HashPair::HashPair(Ad_Object* k, Ad_Object* v) {
	key = k;
	value = v;
}

HashPair::~HashPair() {
	// TODO: implement this
}

Ad_Object* HashPair::GetKey() {
	return key;
}

Ad_Object* HashPair::GetValue() {
	return value;
}